pkg/generate/code/tags.go

```diff
@@ -0,0 +1,154 @@
+//	 		tag := svcapitypes.Tag{Key: &k, Value: &v}
+//	 		result = append(result, &tag)
+//	 		delete(tags, k)
+//	 	}
+//	 }
+//	 for k, v := range tags {
+//	 	tag := svcapitypes.Tag{Key: &k, Value: &v}
+//	 	result = append(result, &tag)
+//	 }
+func GoCodeConvertToACKTags(r *model.CRD, sourceVarName string, targetVarName string, keyOrderVarName string, indentLevel int) string {
+
+	out := "\n"
+	indent := strings.Repeat("\t", indentLevel)
+	tagField, _ := r.GetTagField()
+
+	if tagField == nil {
+		return ""
+	}
+
+	tagFieldShapeType := tagField.ShapeRef.Shape.Type
+	keyMemberName := r.GetTagKeyMemberName()
+	valueMemberName := r.GetTagValueMemberName()
+
+	out += fmt.Sprintf("%sif len(%s) == 0 {\n", indent, sourceVarName)
+	out += fmt.Sprintf("%s\treturn %s, %s\n", indent, targetVarName, keyOrderVarName)
+	out += fmt.Sprintf("%s}\n", indent)
+
+	switch tagFieldShapeType {
+	case "list":
+		out += fmt.Sprintf("%sfor _, t := range %s {\n", indent, sourceVarName)
+		out += fmt.Sprintf("%s\tif t.%s != nil {\n", indent, keyMemberName)
+		out += fmt.Sprintf("%s\t\t%s = append(%s, *t.%s)\n", indent, keyOrderVarName, keyOrderVarName, keyMemberName)
+		out += fmt.Sprintf("%s\t\tif t.%s != nil {\n", indent, valueMemberName)
+		out += fmt.Sprintf("%s\t\t\t%s[*t.%s] = *t.%s\n", indent, targetVarName, keyMemberName, valueMemberName)
+		out += fmt.Sprintf("%s\t\t} else {\n", indent)
+		out += fmt.Sprintf("%s\t\t\t%s[*t.%s] = \"\"\n", indent, targetVarName, keyMemberName)
+		out += fmt.Sprintf("%s\t\t}\n", indent)
+		out += fmt.Sprintf("%s\t}\n", indent)
+		out += fmt.Sprintf("%s}\n", indent)
+
+	case "map":
+		out += fmt.Sprintf("%sfor k, v := range %s {\n", indent, sourceVarName)
+		out += fmt.Sprintf("%s\tif v == nil {\n", indent)
+		out += fmt.Sprintf("%s\t\t%s[k] = \"\"\n", indent, targetVarName)
+		out += fmt.Sprintf("%s\t} else {\n", indent)
+		out += fmt.Sprintf("%s\t\t%s[k] = *v\n", indent, targetVarName)
+		out += fmt.Sprintf("%s\t}\n", indent)
+		out += fmt.Sprintf("%s}\n", indent)
+	default:
+		msg := "error: tag type can only be a list of a map"
+		panic(msg)
+	}
+
+	return out
+}
+
+// GoCodeFromACKTags returns Go code that converts ACKTags
+// to the Resource Tag shape type. Tag fields can only be
+// maps or lists of Tag Go type. If Tag field is a list,
+// when converting from ACK Tags, we try to preserve the 
+// original order
+// 
+//
+//
+// Sample output:
+//
+//	 for _, k := range keyOrder {
+//	 	v, ok := tags[k]
+//	 	if ok {
+//	 		tag := svcapitypes.Tag{Key: &k, Value: &v}
+//	 		result = append(result, &tag)
+//	 		delete(tags, k)
+//	 	}
+//	 }
+//	 for k, v := range tags {
+//	 	tag := svcapitypes.Tag{Key: &k, Value: &v}
+//	 	result = append(result, &tag)
+//	 }
+func GoCodeFromACKTags(r *model.CRD, tagsSourceVarName string, orderVarName string, targetVarName string, indentLevel int) string {
+	out := "\n"
+	indent := strings.Repeat("\t", indentLevel)
+	tagField, _ := r.GetTagField()
+
+	if tagField == nil {
+		return ""
+	}
```

Comment:
```suggestion
	out := "\n"
	indent := strings.Repeat("\t", indentLevel)

	tagField, err := r.GetTagField()
	if err != nil {
	   panic("...")
	}
	if tagField == nil {
		return ""
	}
```

Reply:
I understand that `GetTagField` error is probably handled at some early stages of in code-gen. If we wanted to avoid checking for errors here, need different function signatures and different paradigm. It is always a good practice to check for errors.